stingray/pulse/pulsar.py

```diff
@@ -215,36 +217,57 @@ def phase_exposure(start_time, stop_time, period, nbin=16, gti=None):
 def fold_events(times, *frequency_derivatives, **opts):
     """Epoch folding with exposure correction.
 
+    By default, the keyword `times` accepts a list of
+    unbinned photon arrival times. If the input data is
+    a (binned) light curve, then `times` will contain the
+    time stamps of the observation, and `weights` should
+    be set to the corresponding fluxes or counts.
+
     Parameters
     ----------
     times : array of floats
+        Photon arrival times, or, if `weights` is set,
```

Comment:
What happens if `weights` is set? Don't leave us hanging!

Reply:
I do like a good cliffhanger! :D 

I think I introduced some bugs while I fixed merge conflicts when I merged the current main branch. Should be fixed now.